src/main/java/web/exception/ReservationExceptionAdvice.java

```diff
@@ -0,0 +1,22 @@
+package web.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class ReservationExceptionAdvice {
+
+    @ExceptionHandler(ReservationDuplicateException.class)
+    public ResponseEntity<String> handleReservationDuplicateException(ReservationDuplicateException e) {
+        return ResponseEntity.status(HttpStatus.CONFLICT)
+                .body(e.getMessage());
+    }
+
+    @ExceptionHandler(ReservationNotFoundException.class)
+    public ResponseEntity<String> handleReservationNotFoundException(ReservationNotFoundException e) {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                .body(e.getMessage());
+    }
```

Comment:
전역적인 예외처리 좋습니다! 다만 서비스가 확장됨에 따라 추가되는 exception에 대한 handle 메서드가 계속해서 늘어날 것 같은데요 
exception 각각에 대해서 handler를 구현하기 보단 RoomEscapeException 같은 custom exception 클래스 생성하고 상속시켜서 상위클래스에 대해서만 handler 처리하면 어떨까요?

```suggestion
    @ExceptionHandler(RoomEscapeException.class)
    public ResponseEntity<String> handleRoomEscapeException(RoomEscapeException e) {
        return
            ResponseEntity.status(e.getErrorCode())
            .body(e.getErrorDetail());
    }
```
 
```
public class RoomEscapeException extends RuntimeException {
    private final HttpStatus errorCode;
    private final String detail;

    public RoomEscapeException(HttpStatus errorCode, String detail) {
        this.errorCode = errorCode;
        this.detail = detail;
    }

    public HttpStatus getErrorCode() { return errorCode; }

    public String getErrorDetail() { return detail; }
}
```

```
public class ReservationNotFoundException extends RoomEscapeException {
    public ReservationNotFoundException(long id) {
        super(HttpStatus.NOT_FOUND, "ID가 " + id + "인 예약 정보를 찾을 수 없습니다.");
    }
}
```

Reply:
좋습니다!! `ErrorCode`를 활용해서 `Error`를 구분하기 시작하니까, 세부 `Exception`들(`ReservationNotFoundException`, `ReservationDuplicateException` 등등)도 크게 필요하지 않아졌습니다! 👍👍